code/game/gamemodes/survivor.dm

```diff
@@ -0,0 +1,450 @@
+                break
+
+    // repair dmg
+    if(iswelder(W) && current_hp < max_hp)
+        var/obj/item/tool/weldingtool/WT = W
+        if(WT.remove_fuel(0, user))
+            user.visible_message("<span class='notice'>[user] started repairing \the [src]</span>","<span class='notice'>You started repairing \the [src]</span>")
+            if(do_after(user, 1 SECONDS, TRUE, 5, BUSY_ICON_BUILD,, TRUE))
+                playsound(get_turf(src), 'sound/items/Welder2.ogg', 25, 1)
+                current_hp = min(current_hp + 25, max_hp)
+    else if(iswelder(W) && current_hp == max_hp)
+        to_chat(user, "That doesn't look damaged!")
+
+    // Required Nearby components
+    // TODO: This seems wasteful.
+    var/nearby_setup = 0
+    for (var/R in required_nearby)
+        var/obj/found = locate(R) in range(2, src)
+        if (found && found.anchored == 1)
+            nearby_setup++
+
+    if(length(required_components) == 0 && current_hp == max_hp && nearby_setup == length(required_nearby))
+        activate_beacon(user)
+
+/obj/item/beacon/rescue/proc/activate_beacon(mob/user as mob)
+    icon_state = "[icon_activated]"
+    playsound(src, 'sound/machines/twobeep.ogg', 15, 1)
+    user.visible_message("<span class='notice'>[user] activates [src]</span>", "<span class='notice'>You activate [src]</span>")
+    activated = 1
+    SetLuminosity(1)
+
+    addtimer(CALLBACK(src, .proc/make_noise), 3 SECONDS, TIMER_LOOP)
+    addtimer(CALLBACK(src, .proc/call_distress_team), distress_timer, TIMER_UNIQUE)
+
+/obj/item/beacon/rescue/attack_self(mob/user)
+    if(anchored)
+        to_chat(user, "<span class='warning'>It's already been anchored. Just leave it.</span>")
+        return
+
+    if(!ishuman(user)) 
+        return
+
+    if(!user.mind)
+        to_chat(user, "<span class='warning'>It doesn't seem to do anything for you.</span>")
+        return
+
+    var/area/A = get_area(user)
+    if(A && istype(A) && A.ceiling >= CEILING_METAL)
+        to_chat(user, "<span class='warning'>You have to be outside or under a glass ceiling to activate this.</span>")
+        return
+
+    user.visible_message("<span class='notice'>[user] starts setting up [src] on the ground.</span>",
+    "<span class='notice'>You start setting up [src] on the ground and inputting all the data it needs.</span>")
+    if(do_after(user, activation_time, TRUE, 5, BUSY_ICON_FRIENDLY))
+        user.transferItemToLoc(src, user.loc)
+        anchored = 1
+        w_class = 10
+       
+
+/obj/item/beacon/rescue/proc/make_noise()
+    playsound(loc, 'sound/machines/twobeep.ogg', 25, 1)
+
+/obj/item/beacon/rescue/proc/call_distress_team()
+    SSticker.mode.activate_distress()
+
+
+/*
+SPAWNS
+*/
+
+/obj/effect/landmark/survivor/spawn/human/Initialize()
+    . = ..()
+    GLOB.survivor_spawn_human += loc
+    flags_atom |= INITIALIZED
+    return INITIALIZE_HINT_QDEL
+
+/obj/effect/landmark/survivor/spawn/xeno/Initialize()
```

Comment:
Cant you just use the original xeno spawn landmarks?

Reply:
I could but for now, i just wanted to have my own, this is also the reason i seperated the map out in a seperate commit because i want to redo that last.